src/Entities/ExpOrb.cpp

```diff
@@ -47,33 +50,74 @@ void cExpOrb::SpawnOn(cClientHandle & a_Client)
 void cExpOrb::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 {
 	DetectCacti();
+	m_TicksAlive++;
 
-	// Check player proximity no more than twice per second
-	if ((m_TicksAlive % 10) == 0)
+	bool FindPlayer = false;
+
+	// Find closest player within 6.5 meter (slightly increase detect range to have same effect in client)
+	m_World->DoWithClosestPlayer(GetPosition(), 6.5, [&](cPlayer & a_Player) -> bool
 	{
-		cPlayer * a_ClosestPlayer(m_World->FindClosestPlayer(Vector3f(GetPosition()), 5, false));
-		if ((a_ClosestPlayer != nullptr) && (!a_ClosestPlayer->IsGameModeSpectator()))
+		FindPlayer = true;
+		Vector3f a_PlayerPos(a_Player.GetPosition());
+		a_PlayerPos.y += 0.8f;
+		Vector3f a_Distance(a_PlayerPos - GetPosition());
+		double Distance(a_Distance.Length());
+
+		if (Distance < 0.7f)
 		{
-			Vector3f a_PlayerPos(a_ClosestPlayer->GetPosition());
-			a_PlayerPos.y++;
-			Vector3f a_Distance(a_PlayerPos - GetPosition());
-			double Distance(a_Distance.Length());
-			if (Distance < 0.5f)
-			{
-				LOGD("Player %s picked up an ExpOrb. His reward is %i", a_ClosestPlayer->GetName().c_str(), m_Reward);
-				a_ClosestPlayer->DeltaExperience(m_Reward);
+			LOGD("Player %s picked up an ExpOrb. His reward is %i", a_Player.GetName().c_str(), m_Reward);
+			a_Player.DeltaExperience(m_Reward);
 
-				m_World->BroadcastSoundEffect("entity.experience_orb.pickup", GetPosition(), 0.5f, (0.75f + (static_cast<float>((GetUniqueID() * 23) % 32)) / 64));
+			m_World->BroadcastSoundEffect("entity.experience_orb.pickup", GetPosition(), 0.5f, (0.75f + (static_cast<float>((GetUniqueID() * 23) % 32)) / 64));
 
-				Destroy(true);
-				return;
+			Destroy(true);
+			return true;
+		}
+
+		/** Experience orb will "float" or glide toward the player up to a distance of 6 blocks,
+		speeding up as they get nearer to the player, Speed range 6 - 12 m per second, accelerate 60 m per second^2 */
+		if (MoveToPlayer)  // Already flying to player
+		{
+			Vector3d SpeedDelta(a_Distance);
+			SpeedDelta.Normalize();
+			SpeedDelta *= 3;
+
+			Vector3d CurrentSpeed = GetSpeed();
+
+			CurrentSpeed += SpeedDelta;
+
```

Comment:
Too many empty lines with no effect. Only use empty lines for blocks of code for which it makes sense to add a comment:
```cpp
PreviousCodeBlock;

// What the next block does:
CodeBlock;
CodeBlockContinued;
MoreCode;
...
```